src/main/minio.js

```diff
@@ -2036,6 +2038,63 @@ export class Client {
         .on('end', () => cb(null, bucketNotification))
     })
   }
+
+  // Listens for bucket notifications. Returns an EventEmitter.
+  listenBucketNotification(bucketName, notificationARN) {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError(`Invalid bucket name: ${bucketName}`)
+    }
+    if (typeof notificationARN !== 'string') {
+      throw new TypeError('notificationARN must be of type string')
+    }
+
+    let emitter = new EventEmitter()
+
+    // Listening constitutes repeatedly requesting s3 whether or not any
+    // changes have occurred. If we receive an 'end' event, stop.
+
+    let end = false
+    emitter.on('end', () => { end = true })
```

Comment:
I don't think `emitter.removeListener()` is idiomatic to stop notifications — that's just to remove the listener for notifications. Did you pick that up from streams?

And since we start the polling loop with `process.nextTick`, all listeners should be attached by the time we start the GET request.